### PR TITLE
Removed unnecessary dependency of Logging.ApplicationInsights on AspNetCore.Mvc

### PR DESCRIPTION
**Pull request by tompazourek**
We're using the "Microsoft.Azure.WebJobs.Logging.ApplicationInsights" package on an ASP.NET MVC 5 project and these MVC Core references are causing some trouble (due to mixing MVC and MVC Core together).

It seems that they are not actually necessary, all it needs is the HTTP abstractions, so I removed those dependencies.

### CODE REVIEW

**Comment by tompazourek** [comment]
There seems to be an issue on the build server [**azure-webjobs-sdk (UT)**](https://ci.appveyor.com/project/appsvc/azure-webjobs-sdk-881mw/history). All the builds fail for some unrelated reason (missing bin path).

I do not believe it's related to my changes.

**Comment by tompazourek** [comment]
@brettsam Is there any chance to get some feedback on this, please? It's been a while... Thank you.

**Comment by jvilimek** [comment]
Anything new?

**Comment by footcha** [comment]
We are suffering from the same issue. Is there a plan to fix it, please?

**Comment by jvilimek** [comment]
@brettsam ?

**Comment by jvilimek** [comment]
@fabiocav ?

**Comment by brettsam** [comment]
Sorry for missing this. @lmolkova -- any reason we still need to have these references?

**Comment by lmolkova** [comment]
We do not need these references to my best knowledge. They've been there for [quite a while](https://github.com/Azure/azure-webjobs-sdk/commit/5140983cb163b20bf6af60afccebb705bc80ad80#diff-6982015521b45c0bdd5b3940665d9d62) and are not ApplicationInsights SDK dependencies

**Comment by jvilimek** [comment]
Thanks for comment/review. So.. Would it be possible to merge? We would really appreciate it!

**Comment by tompazourek** [comment]
@lmolkova Thank you for checking this, that's great news 🎉 I'm also still very keen to see this one merged.

**Comment by tompazourek** [comment]
@brettsam @fabiocav Are there any obstacles to get this one merged?

**Comment by brettsam** [comment]
It makes me nervous whenever we remove dependencies b/c that has an impact on Functions (we've broken people accidentally before). I wanted to make sure we were tracking this so I added a work item here: https://github.com/Azure/azure-functions-host/issues/6636. I feel more comfortable merging this now.

**Comment by tompazourek** [comment]
@brettsam Yeah, I can definitely relate to that. Thanks for merging this. 👍 

May I ask how does the release process work here for Microsoft.Azure.WebJobs? On NuGet, I see the latest is 3.0.18 from August, but here in [GitHub releases](https://github.com/Azure/azure-webjobs-sdk/releases) I see 3.0.18 released in June, and 3.0.19 released in July, and then also 3.0.20 and 3.0.22... It's a bit confusing because of the disparities between releases and what's actually on NuGet.

Basically, what I'm just wondering about is when I could expect to see a NuGet package containing these changes -- a week from now, a month from now, several months from now?

**Comment by jvilimek** [comment]
@brettsam thanks & appreciated the merge!

**Comment by tompazourek** [comment]
@brettsam It's been a few weeks since the merge, so I'm asking again. Any information about **when we could see a release of 
Microsoft.Azure.WebJobs.Logging.ApplicationInsights** containing these changes? Thank you.

**Comment by brettsam** [comment]
I've just pushed it this morning -- https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/3.0.23.

**Comment by brettsam** [comment]
Well, shoot, scratch that. I inadvertently pushed version 3.0.23, which isn't deployed to production yet (and the corresponding WebJobs dependency isn't ready). It should be all ready in a couple of weeks and I'll work to get this hidden in the meantime.

**Comment by tompazourek** [comment]
@brettsam Thank you, appreciate that. These dependencies can be difficult 😅 Looking forward to that version once it's ready.

**Comment by tompazourek** [comment]
@brettsam Sorry to bother you again, it's been almost another month now. I'm just curious if there are any news regarding this. I read about the #2600 issue and that the logging package ended up hidden because the main one wasn't out yet.

**Any idea when we could see the 3.0.23 release on NuGet?** Thanks.

**EDIT:** Released already (Nov 9th 2020), thanks, everyone.
